.github/workflows/build.yml

```diff
@@ -55,3 +62,33 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       - name: Remove installed project artifacts
         run: mvn dependency:purge-local-repository -DmanualInclude='${project.groupId}' -DresolutionFuzziness=groupId
+  deploy:
+    if: github.ref == 'refs/heads/master'
+    needs: build
+    runs-on: ubuntu-24.04
+    environment: maven-deploy
```

Comment:
The four secrets used below are restricted to this environment, and only available on `master`. This should prevent exfiltration.